include/nds/arm9/teak/twlwram.h

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: Zlib
+//
+// Copyright (C) 2020 Gericom
+
+#ifndef LIBNDS_NDS_ARM9_TEAK_TWLWRAM_H__
```

Comment:
The nocash-proposed term here is `New Shared WRAM`, or `nwram` for short. I would standardize on that.

Also, it is not a Teak-specific functionality. I would move this out of `arm9/teak`.

*Also* (sorry for being pedantic), the NWRAM registers also exist on ARM7. WRAM A/B/C allocation is read-only, but address mapping (MBK6-8) are read-write and separate on ARM7.

It is also important to note that some homebrew launching methods do not support changing SCFG flags and may not support, for example, NWRAM, as a result.

Essentially, the changes I would do here are as follows:

* define a `common/nwram.h` and rename all occurences of `TWR_WRAM`, `TWLWRAM`, etc. to `NWRAM`
* Provide all defines and `nwramGetBlockAddress` and `nwramSetBlockMapping` (naming more consistent with libnds convention) on both ARM9 and ARM7,
* Provide `nwramMapSlotA/SlotB/SlotC` on ARM9 only.

Reply:
I've done everything except for the renaming of the MBK registers. I'm not sure about whether that's a good idea or not, given that GBATEK is the only reference we have, and it uses said names. This could be done in the future, anyway.

Reply:
At the very least, I would keep both around as synonyms. It makes code more readable.

Reply:
I've created this issue: https://github.com/blocksds/sdk/issues/92 I wouldn't want to do it as part of this PR, it is quite complex already.